Failure to build with notify 5.0.0-pre.11

It was working file with notify 5.0.0-pre.10, but with .11 I get:

```
error[E0599]: no function or associated item named `new_immediate` found for trait object `dyn Watcher` in the current scope
  --> /Users/me/.cargo/registry/src/github.com-1ecc6299db9ec823/linemux-0.2.2/src/events.rs:62:64
   |
62 |         let inner: notify::RecommendedWatcher = NotifyWatcher::new_immediate(move |res| {
   |                                                                ^^^^^^^^^^^^^ function or associated item not found in `dyn Watcher`
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](http://semver.org).
 
 ## [Unreleased]
 
+### Fixed
+- Update `notify` to `5.0.0-pre.11` to fix build errors
+
 ## [0.2.2] - 2021-06-06
 
 ### Fixed

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,7 +21,7 @@ tokio = ["tokio_"]
 
 [dependencies]
 futures-util = { version = "0.3", default-features = false, features = ["std"] }
-notify = "5.0.0-pre.9"
+notify = "5.0.0-pre.11"
 pin-project-lite = "0.1"
 tokio_ = { package = "tokio", version = "1.0", features = ["fs", "io-util", "sync", "time"], optional = true }
 

--- a/src/events.rs
+++ b/src/events.rs
@@ -196,19 +196,18 @@ impl MuxedEvents {
     }
 
     fn handle_event(&mut self, event: &mut notify::Event) {
-        use std::mem;
-
-        let mut paths = mem::replace(&mut event.paths, Vec::new());
+        let paths = &mut event.paths;
+        let event_kind = &event.kind;
 
         // TODO: properly handle any errors encountered adding/removing stuff
         paths.retain(|path| {
             // Fixes a potential race when detecting file rotations.
             let path_exists =
-                if let notify::EventKind::Remove(notify::event::RemoveKind::File) = &event.kind {
+                if let notify::EventKind::Remove(notify::event::RemoveKind::File) = &event_kind {
                     false
                 } else if let notify::EventKind::Modify(notify::event::ModifyKind::Name(
                     notify::event::RenameMode::From,
-                )) = &event.kind
+                )) = &event_kind
                 {
                     if cfg!(target_os = "macos") {
                         path.exists()
@@ -234,8 +233,6 @@ impl MuxedEvents {
 
             self.watched_files.contains(path)
         });
-
-        let _ = mem::replace(&mut event.paths, paths);
     }
 
     fn __poll_next_event(

--- a/src/events.rs
+++ b/src/events.rs
@@ -59,7 +59,7 @@ impl MuxedEvents {
     /// Constructs a new `MuxedEvents` instance.
     pub fn new() -> io::Result<Self> {
         let (tx, rx) = mpsc::unbounded_channel();
-        let inner: notify::RecommendedWatcher = NotifyWatcher::new_immediate(move |res| {
+        let inner: notify::RecommendedWatcher = notify::RecommendedWatcher::new(move |res| {
             // The only way `send` can fail is if the receiver is dropped,
             // and `MuxedEvents` controls both. `unwrap` is not used,
             // however, since `Drop` idiosyncrasies could otherwise result
@@ -86,13 +86,13 @@ impl MuxedEvents {
             || self.watched_directories.contains_key(&path.to_path_buf())
     }
 
-    fn watch(watcher: &mut notify::RecommendedWatcher, path: impl AsRef<Path>) -> io::Result<()> {
+    fn watch(watcher: &mut notify::RecommendedWatcher, path: &Path) -> io::Result<()> {
         watcher
             .watch(path, notify::RecursiveMode::NonRecursive)
             .map_err(notify_to_io_error)
     }
 
-    fn unwatch(watcher: &mut notify::RecommendedWatcher, path: impl AsRef<Path>) -> io::Result<()> {
+    fn unwatch(watcher: &mut notify::RecommendedWatcher, path: &Path) -> io::Result<()> {
         watcher.unwatch(path).map_err(notify_to_io_error)
     }
 

--- a/src/reader.rs
+++ b/src/reader.rs
@@ -231,7 +231,7 @@ impl MuxedLines {
                             if event.paths.is_empty() {
                                 (StreamState::Events, None)
                             } else {
-                                let paths = std::mem::replace(&mut event.paths, Vec::new());
+                                let paths = std::mem::take(&mut event.paths);
                                 (StreamState::ReadLines(paths, 0), None)
                             }
                         }

